feature-charity/src/main/java/com/hedvig/android/feature/charity/CharityActivity.kt

```diff
@@ -0,0 +1,38 @@
+package com.hedvig.android.feature.charity
+
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.material3.windowsizeclass.ExperimentalMaterial3WindowSizeClassApi
+import androidx.compose.material3.windowsizeclass.calculateWindowSizeClass
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.core.view.WindowCompat
+import coil.ImageLoader
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.feature.charity.ui.CharityScreen
+import org.koin.android.ext.android.get
+import org.koin.androidx.viewmodel.ext.android.getViewModel
+
+@OptIn(ExperimentalMaterial3WindowSizeClassApi::class)
+class CharityActivity : AppCompatActivity() {
```

Comment:
Should this extend `BaseActivity`?

Reply:
A good opportunity to talk about this.
We don't have access to BaseActivity from other modules since it's inside :app. So either I'll extract it out, or consider what is it that we're doing inside BaseActivity and see if we need it or if we can do it in a different way.

As I see it now, it's there to share the behavior of listening to locale changes. Could I get some historical information of how this works and helps in these situations? I know we have a fixed language picked from the settings, and if we're in for example this screen there's no way to change that language picked without leaving it, going into settings, changing the language, and then navigating back to this screen after we've already changed the language therefore it should just be applied.
I'm probably missing smth since I haven't really worked with Android locales and stuff like that so please do help me out here :D

Reply:
It's the thing that does basically what the new APIs for changing language does - it restarts every activity when the language changes, so that these changes are picked up.

Reply:
So in the context of the CharityActivity screen, is it possible that the language will ever change while this screen is showing?
Is this needed in the first place because we specifically allow people to change their language from inside the app? (something which is not that common I guess)
Is there a way to do this without using this BaseActivity inheritance? And if not, we should still be able to extract this from inside the :app module in order for other modules to have access to it right?

Reply:
I'd say it's very unlikely, but one possible path would be:

1. User opens the app
2. User navigates to charity screen
3. User backgrounds the app
4. User opens the app through a chat notification
5. User accesses settings through the chat screen
6. User changes language in this screen
7. User navigates back twice, landing on a now incorrectly localized charity screen

Reply:
Yup makes sense, let's see how it goes with the thing you're trying out with the language APIs then and go from there.
In the meantime I guess I'll have to
1) make BaseActivity available to this module in order to merge it in
2) Take the key gear branch and take it off of the top of this one and merge that into develop to allow us to do other stuff in the meantime while keeping this branch on hold until this is fixed
3) Just merge this in and consider this something that we can fix ASAP afterwards, since the chances that someone encounters this and is bothered by it seem extremely low. And the impact isn't the end of the world either.

If I am to do one, this is the order of the one I want to do the most, to the least 3 -> 1 -> 2 but you can convince me otherwise 😅